cli/actions/resource.go

```diff
@@ -0,0 +1,67 @@
+package actions
+
+import (
+	"context"
+	"errors"
+
+	"github.com/kubeshop/tracetest/cli/openapi"
+	"go.uber.org/zap"
+)
+
+type ApplyArgs struct {
+	File string
+}
+
+type ResourceActions interface {
+	Apply(ctx context.Context, args ApplyArgs) error
+	List(ctx context.Context) error
+	Export(ctx context.Context, ID string) error
+	Delete(ctx context.Context, ID string) error
+}
+
+type resourceArgs struct {
+	logger *zap.Logger
+	client *openapi.APIClient
+}
+
+type resourceArgsOption = func(*resourceArgs)
+
+type ResourceRegistry map[string]ResourceActions
+type supportedResources string
```

Comment:
Since we are typing resource names as `supportedResources`, we could turn this type public and define the `ResourceRegistry` as `map[supportedResources]ResourceActions`, does it make sense?

A suggestion of refactor:

```suggestion
type ResourceRegistry map[SupportedResource]ResourceActions
type SupportedResource string
```